File: codes/ruby/chapter05/answer_A26.rb
def is_prime(n)
  limit = Math.sqrt(n).to_i
  (2..limit).each do |i|
    return false if n % i == 0
  end
  return true
end


# 入力
query_count = gets.to_i
query_values = Array.new(query_count)
for index in 0...query_count
  query_values[index] = gets.to_i
end

# 出力
query_values.each do |value|
  if is_prime(value)
    puts "Yes"
  else
    puts "No"
  end
end

File: codes/elixir/chapter05/answer_A26.exs
defmodule Prime do
  def is_prime(n) do
    limit = :math.sqrt(n) |> round() |> :erlang.trunc()
    Enum.all?(2..limit, fn i -> rem(n, i) != 0 end)
  end
end

n = String.trim(IO.gets("")) |> String.to_integer
IO.puts Prime.is_prime(n)
